eZ/Publish/Core/MVC/Symfony/Controller/Content/QueryController.php

```diff
@@ -81,4 +88,67 @@ private function runQuery(ContentView $view, $method)
         );
         $view->addParameters([$view->getParameter('query')['assign_results_to'] => $searchResults]);
     }
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\View\ContentView $view
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @return \eZ\Publish\Core\MVC\Symfony\View\ContentView
+     */
+    public function pagingQueryAction(ContentView $view, Request $request)
+    {
+        $this->runPagingQuery($view, $request);
+
+        return $view;
+    }
+
+    /**
+     * @param \eZ\Publish\Core\MVC\Symfony\View\ContentView $view
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     */
+    private function runPagingQuery(ContentView $view, Request $request)
+    {
+        $queryParameters = $view->getParameter('query');
+
+        $limit = $this->getParameter('limit', 10, $queryParameters);
+        $page_param = $this->getParameter('page_param', 'page', $queryParameters);
+
+        $page = $request->get($page_param, 1);
```

Comment:
... I rather expected here page number instead of page parameter name. Why so much trouble?

Reply:
This was done to address @andrerom  first comment. The idea here was to be ready when multiple query will be possible.